lib/src/domain/communication/api_exception.dart

```diff
@@ -0,0 +1,15 @@
+class ApiException implements Exception {
+  final int code;
+
+  ApiException([this.code]);
```

Comment:
teraz to nigdzie nie jest wykorzystwany statusCode - mimo, że notFound i unauthorized mają jasno określone statusy błędu, imo można wywalić, a jak będą do czegoś potrzebne to zawsze można dodać, ale jak uważasz, jak wolisz, żeby był to warto dodać do wspomnianych wyżej exceptionów określony kod :)

Reply:
też to zauważyłem